pulsar/impl_consumer.go

```diff
@@ -154,27 +155,47 @@ func (c *consumer) Unsubscribe() error {
 }
 
 func (c *consumer) Receive(ctx context.Context) (Message, error) {
+	if len(c.consumers) > 1 {
+		for _, pc := range c.consumers {
+			go func(pc Consumer) {
+				for {
+					msg, err := pc.Receive(ctx)
```

Comment:
Please follow java/cpp logic, load c.queue in background, not at the time receive is called.